### PR TITLE
Extended JUnit output

### LINKED ISSUE #120: JUnit Style xml output file including the number of CHECKS
Since the amount of checks is an important metrics, propose to give out this metric in the XML output file.

### PR DESCRIPTION
**Pull request by itavero**
Added the following to the JUnit XML output:
* Location of test (path to file and line number)
* Number of checks (assertions) in a test case

Initial added this while working on #967, but my daytime job is taking up a lot of my time and these changes have been requested in other issues too, hence the PR.

This should fix #120 and #903.

### CODE REVIEW

**Comment by itavero** [comment]
I'd love to hear your opinions, @basvodde & @arstrube...

File: tests/JUnitOutputTest.cpp [hunk 1]
```diff
@@ -499,7 +520,7 @@ TEST(JUnitOutputTest, testFailureWithDifferentFileAndLine)
 {
     testCaseRunner->start()
             .withGroup("testGroupWithFailingTest")
-                .withTest("FailingTestName").thatFails("Test failed", "importantFile", 999)
+            .withTest("FailingTestName", "sourcefile.c", 123).thatFails("Test failed", "importantFile", 999)
```
Comment (basvodde) [inline tests/JUnitOutputTest.cpp]: This change isn't too good. It is breaking the builder. The way you would probably want to get this is:

.withTest("FailingTestName").in("sourcefile.c").lineNumber(123)

That way you do not need to pollute all the tests with the "sourcefile.c" information, which is irrelevant for most of the tests.

Making builders work like that is fun :)

Comment (itavero) [inline tests/JUnitOutputTest.cpp]: I'll look into it. Never really wrote a builder like this. :wink:

**Comment by itavero** [comment]
@basvodde Besides the changes I made to the unit tests (`tests/JUnitOutputTest.cpp`), do the other changes look good to you?

**Comment by itavero** [comment]
For some reason the Windows 64 MinGW build failed, but I do not really see why...

**Comment by Andne** [comment]
Every so often the install command fails, never really been sure why, seems like there's a network glitch occasionally that prevents the download or something.

**Comment by itavero** [comment]
@Andne, is there an easy way to restart the build without pushing a new commit?
And has an issue been created to make the build more reliable?

**Comment by arstrube** [comment]
@itavero close and reopen the PR.

**Comment by itavero** [comment]
You have got to be kidding me.. now the 32bit Windows MinGW build fails... :worried:

**Comment by basvodde** [comment]
Nice.

**Pull request merged**